### ISSUE TITLE
Failed to update

### ISSUE THREAD

**Issue by Yaruze66** [Comfy-Org/ComfyUI-Manager#228]
Now when I use the "Update All" button, I start getting the warning: "Failed to update ComfyUI or several extensions / TypeError: Cannot set properties of null (setting 'onclick')", but as far as I understand, ComfyUI itself and all my extensions still keep updating and working properly.

Console log:
<details>

```

** ComfyUI start up time: 2023-12-08 22:51:28.585658

#######################################################################
[ComfyUI-Manager] Starting dependency installation/(de)activation for the extension

Install: pip packages for 'D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Impact-Pack'
Install: install script for 'D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Impact-Pack'
 ### ComfyUI-Impact-Pack: Check dependencies
 ### ComfyUI-Impact-Pack: Updating subpack
 [Impact Pack] EXECUTE: ['D:\\ComfyUI_windows_portable\\python_embeded\\python.exe', 'install.py'] in 'D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\impact_subpack'
 req_path: D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\impact_subpack\requirements.txt
 req_path: D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\impact_subpack\requirements.txt
 req_path: D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\requirements.txt
 ### ComfyUI-Impact-Pack: Check basic models
Install: pip packages for 'D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfyui_controlnet_aux'
 Requirement already satisfied: importlib_metadata in d:\comfyui_windows_portable\python_embeded\lib\site-packages (6.8.0)
 Requirement already satisfied: zipp>=0.5 in d:\comfyui_windows_portable\python_embeded\lib\site-packages (from importlib_metadata) (3.17.0)
 Requirement already satisfied: huggingface_hub in d:\comfyui_windows_portable\python_embeded\lib\site-packages (0.19.4)
 Requirement already satisfied: filelock in d:\comfyui_windows_portable\python_embeded\lib\site-packages (from huggingface_hub) (3.13.1)
 Requirement already satisfied: fsspec>=2023.5.0 in d:\comfyui_windows_portable\python_embeded\lib\site-packages (from huggingface_hub) (2023.10.0)
 Requirement already satisfied: requests in d:\comfyui_windows_portable\python_embeded\lib\site-packages (from huggingface_hub) (2.31.0)
 Requirement already satisfied: tqdm>=4.42.1 in d:\comfyui_windows_portable\python_embeded\lib\site-packages (from huggingface_hub) (4.66.1)
 Requirement already satisfied: pyyaml>=5.1 in d:\comfyui_windows_portable\python_embeded\lib\site-packages (from huggingface_hub) (6.0.1)
 Requirement already satisfied: typing-extensions>=3.7.4.3 in d:\comfyui_windows_portable\python_embeded\lib\site-packages (from huggingface_hub) (4.8.0)
 Requirement already satisfied: packaging>=20.9 in d:\comfyui_windows_portable\python_embeded\lib\site-packages (from huggingface_hub) (23.2)
 Requirement already satisfied: colorama in d:\comfyui_windows_portable\python_embeded\lib\site-packages (from tqdm>=4.42.1->huggingface_hub) (0.4.6)
 Requirement already satisfied: charset-normalizer<4,>=2 in d:\comfyui_windows_portable\python_embeded\lib\site-packages (from requests->huggingface_hub) (3.3.2)
 Requirement already satisfied: idna<4,>=2.5 in d:\comfyui_windows_portable\python_embeded\lib\site-packages (from requests->huggingface_hub) (3.4)
 Requirement already satisfied: urllib3<3,>=1.21.1 in d:\comfyui_windows_portable\python_embeded\lib\site-packages (from requests->huggingface_hub) (1.26.18)
 Requirement already satisfied: certifi>=2017.4.17 in d:\comfyui_windows_portable\python_embeded\lib\site-packages (from requests->huggingface_hub) (2023.11.17)
Install: pip packages for 'D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Inspire-Pack'
Install: pip packages for 'D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager'
Install: pip packages for 'D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\was-node-suite-comfyui'
 Collecting git+https://github.com/WASasquatch/img2texture.git
   Cloning https://github.com/WASasquatch/img2texture.git to d:\comfyui_windows_portable\comfyui\custom_nodes\was-node-suite-comfyui\tmp\pip-req-build-pi3ar78h
[!]   Running command git clone --filter=blob:none --quiet https://github.com/WASasquatch/img2texture.git 'D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\was-node-suite-comfyui\tmp\pip-req-build-pi3ar78h'
   Resolved https://github.com/WASasquatch/img2texture.git to commit a546609a915caf1dcb84a15bf44b5b639ad924e6
   Preparing metadata (setup.py): started
   Preparing metadata (setup.py): finished with status 'done'
 Requirement already satisfied: pillow<10,>=9.2 in d:\comfyui_windows_portable\python_embeded\lib\site-packages (from img2texture==1.0.6) (9.5.0)
 Collecting git+https://github.com/WASasquatch/cstr
   Cloning https://github.com/WASasquatch/cstr to d:\comfyui_windows_portable\comfyui\custom_nodes\was-node-suite-comfyui\tmp\pip-req-build-n6v8t6kf
[!]   Running command git clone --filter=blob:none --quiet https://github.com/WASasquatch/cstr 'D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\was-node-suite-comfyui\tmp\pip-req-build-n6v8t6kf'
   Resolved https://github.com/WASasquatch/cstr to commit 0520c29a18a7a869a6e5983861d6f7a4c86f8e9b
   Preparing metadata (setup.py): started
   Preparing metadata (setup.py): finished with status 'done'
 Requirement already satisfied: opencv-python-headless<=4.7.0.72 in d:\comfyui_windows_portable\python_embeded\lib\site-packages (from opencv-python-headless[ffmpeg]<=4.7.0.72) (4.7.0.72)
 Requirement already satisfied: numpy>=1.21.2 in d:\comfyui_windows_portable\python_embeded\lib\site-packages (from opencv-python-headless<=4.7.0.72->opencv-python-headless[ffmpeg]<=4.7.0.72) (1.26.2)
[!] WARNING: opencv-python-headless 4.7.0.72 does not provide the extra 'ffmpeg'
 Collecting git+https://github.com/WASasquatch/ffmpy.git
   Cloning https://github.com/WASasquatch/ffmpy.git to d:\comfyui_windows_portable\comfyui\custom_nodes\was-node-suite-comfyui\tmp\pip-req-build-l3x6swrh
[!]   Running command git clone --filter=blob:none --quiet https://github.com/WASasquatch/ffmpy.git 'D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\was-node-suite-comfyui\tmp\pip-req-build-l3x6swrh'
   Resolved https://github.com/WASasquatch/ffmpy.git to commit f000737698b387ffaeab7cd871b0e9185811230d
   Preparing metadata (setup.py): started
   Preparing metadata (setup.py): finished with status 'done'
Install: pip packages for 'D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-WD14-Tagger'
Install: pip packages for 'D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\efficiency-nodes-comfyui'
Install: pip packages for 'D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\rgthree-comfy'

[ComfyUI-Manager] Startup script completed.
#######################################################################


Prestartup times for custom nodes:
   0.0 seconds: D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\rgthree-comfy
  29.9 seconds: D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager

Total VRAM 8192 MB, total RAM 32705 MB
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 2070 SUPER : cudaMallocAsync
VAE dtype: torch.float16
Using pytorch cross attention
Adding extra search path checkpoints D:/stable-diffusion-webui/models/Stable-diffusion
Adding extra search path configs D:/stable-diffusion-webui/models/Stable-diffusion
Adding extra search path vae D:/stable-diffusion-webui/models/VAE
Adding extra search path loras D:/stable-diffusion-webui/models/Lora
Adding extra search path loras D:/stable-diffusion-webui/models/LyCORIS
Adding extra search path upscale_models D:/stable-diffusion-webui/models/ESRGAN
Adding extra search path upscale_models D:/stable-diffusion-webui/models/RealESRGAN
Adding extra search path upscale_models D:/stable-diffusion-webui/models/SwinIR
Adding extra search path embeddings D:/stable-diffusion-webui/embeddings
Adding extra search path hypernetworks D:/stable-diffusion-webui/models/hypernetworks
Adding extra search path controlnet D:/stable-diffusion-webui/models/ControlNet
### Loading: ComfyUI-Impact-Pack (V4.38.2)
### Loading: ComfyUI-Impact-Pack (Subpack: V0.3.2)
### Loading: ComfyUI-Inspire-Pack (V0.47.3)
### Loading: ComfyUI-Manager (V1.10)
### ComfyUI Revision: 1795 [a4ec54a4] | Released on '2023-12-08'
[comfyui_controlnet_aux] | INFO -> Using ckpts path: D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfyui_controlnet_aux\ckpts
[comfyui_controlnet_aux] | INFO -> Using symlinks: False
[comfyui_controlnet_aux] | INFO -> Using ort providers: ['CUDAExecutionProvider', 'DirectMLExecutionProvider', 'OpenVINOExecutionProvider', 'ROCMExecutionProvider', 'CPUExecutionProvider']
D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfyui_controlnet_aux\node_wrappers\dwpose.py:25: UserWarning: DWPose: Onnxruntime not found or doesn't come with acceleration providers, switch to OpenCV with CPU device. DWPose might run very slowly
  warnings.warn("DWPose: Onnxruntime not found or doesn't come with acceleration providers, switch to OpenCV with CPU device. DWPose might run very slowly")
Using pytorch cross attention
Efficiency Nodes: Attempting to add Control Net options to the 'HiRes-Fix Script' Node (comfyui_controlnet_aux add-on)...Success!

[rgthree] Loaded 18 exciting nodes.
[rgthree] Optimizing ComfyUI recursive execution. If queueing and/or re-queueing seems broken, change "patch_recursive_execution" to false in rgthree_config.json

WAS Node Suite: BlenderNeko's Advanced CLIP Text Encode found, attempting to enable `CLIPTextEncode` support.
WAS Node Suite: `CLIPTextEncode (BlenderNeko Advanced + NSP)` node enabled under `WAS Suite/Conditioning` menu.
WAS Node Suite: OpenCV Python FFMPEG support is enabled
WAS Node Suite Warning: `ffmpeg_bin_path` is not set in `D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\was-node-suite-comfyui\was_suite_config.json` config file. Will attempt to use system ffmpeg binaries if available.
WAS Node Suite: Finished. Loaded 198 nodes successfully.

        "Art is the triumph over chaos." - John Cheever


Import times for custom nodes:
   0.0 seconds: D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI_Noise
   0.0 seconds: D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI_ADV_CLIP_emb
   0.0 seconds: D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI_TiledKSampler
   0.0 seconds: D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Custom-Scripts
   0.0 seconds: D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI_UltimateSDUpscale
   0.0 seconds: D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\efficiency-nodes-comfyui
   0.0 seconds: D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\rgthree-comfy
   0.0 seconds: D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-WD14-Tagger
   0.0 seconds: D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Inspire-Pack
   0.1 seconds: D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI_smZNodes
   0.3 seconds: D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager
   0.4 seconds: D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfyui_controlnet_aux
   0.7 seconds: D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Impact-Pack
   1.3 seconds: D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\was-node-suite-comfyui

Starting server

To see the GUI go to: http://127.0.0.1:8188
FETCH DATA from: D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\extension-node-map.json
FETCH DATA from: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/custom-node-list.json
Fetching done.
FETCH DATA from: D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\.cache\1514988643_custom-node-list.json
Fetching done.
FETCH DATA from: D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\.cache\1514988643_custom-node-list.json
Update check done.
FETCH DATA from: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/extension-node-map.json
Update ComfyUI
Update ComfyUI
FETCH DATA from: D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\.cache\1514988643_custom-node-list.json
Update done.
```
</details>

**Comment by uniqornate** [comment]
same here. also it started with "ip adapter model loader". the loader in't work. i mean it's not open the list when i click. just says "null" and "undefined"

**Comment by ltdrdata** [comment]
The issue was not with the update functionality itself, but rather with displaying the RESTART button after the update was completed, causing an error. I have just fixed it. Please try updating again.

**Comment by ETCore7** [comment]
FYI, I just installed ComfyUI Manager. First run: "ComfyUI is outdated!!" Clicking update: "Failed to update ComfyUI". Update All results in a fail too, plus a "see server log". Serverlog: "The installed ComfyUI does not have a Git repository."

(To add to this, I just followed the instructions. CMD box, cd .../custom_nodes, 'git clone https://github.com/ltdrdata/ComfyUI-Manager.git' )

**Comment by ltdrdata** [comment]
> FYI, I just installed ComfyUI Manager. First run: "ComfyUI is outdated!!" Clicking update: "Failed to update ComfyUI". Update All results in a fail too, plus a "see server log". Serverlog: "The installed ComfyUI does not have a Git repository."
> 
> (To add to this, I just followed the instructions. CMD box, cd .../custom_nodes, 'git clone https://github.com/ltdrdata/ComfyUI-Manager.git' )

Goto `ComfyUI` dir and execute `git pull`
What message is displayed?

Is `ComfyUI\.git ` dir exists?

`"The installed ComfyUI does not have a Git repository."` means that it is not the **ComfyUI-Manager** that is installed incorrectly, but the **ComfyUI** itself.

**Comment by rachelcenter** [comment]
i had comfyui installed on an external hard drive and it was all up to date today. then i was messing with deleting a file on the external hard drive and effed up the drive and it became unwrittable so i moved my entire comfy directory off of the external hard drive and put it on an SSD drive. now when I try to load comfy from the SSD, manager says "ComfyUI Manager "Your ComfyUI is too OUTDATED!!!" .... HOW?!?! it was up to date today..... and terminal says "ComfyUI update fail: The installed ComfyUI does not have a Git repository." how do i fix this on mac m2 silicone?

**Comment by rachelcenter** [comment]
> > FYI, I just installed ComfyUI Manager. First run: "ComfyUI is outdated!!" Clicking update: "Failed to update ComfyUI". Update All results in a fail too, plus a "see server log". Serverlog: "The installed ComfyUI does not have a Git repository."
> > (To add to this, I just followed the instructions. CMD box, cd .../custom_nodes, 'git clone https://github.com/ltdrdata/ComfyUI-Manager.git' )
> 
> Goto `ComfyUI` dir and execute `git pull` What message is displayed?
> 
> Is `ComfyUI\.git ` dir exists?
> 
> `"The installed ComfyUI does not have a Git repository."` means that it is not the **ComfyUI-Manager** that is installed incorrectly, but the **ComfyUI** itself.

Last login: Mon Aug 19 21:34:55 on ttys000
(base) rachel@RachelsacStudio ~ % cd /Volumes/RKNavy001/ComfyUI 
(base) rachel@RachelsacStudio ComfyUI % git pull
fatal: not a git repository (or any parent up to mount point /Volumes)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).

**Comment by ltdrdata** [comment]
> > > FYI, I just installed ComfyUI Manager. First run: "ComfyUI is outdated!!" Clicking update: "Failed to update ComfyUI". Update All results in a fail too, plus a "see server log". Serverlog: "The installed ComfyUI does not have a Git repository."
> > > (To add to this, I just followed the instructions. CMD box, cd .../custom_nodes, 'git clone https://github.com/ltdrdata/ComfyUI-Manager.git' )
> > 
> > 
> > Goto `ComfyUI` dir and execute `git pull` What message is displayed?
> > Is `ComfyUI\.git ` dir exists?
> > `"The installed ComfyUI does not have a Git repository."` means that it is not the **ComfyUI-Manager** that is installed incorrectly, but the **ComfyUI** itself.
> 
> Last login: Mon Aug 19 21:34:55 on ttys000 (base) rachel@RachelsacStudio ~ % cd /Volumes/RKNavy001/ComfyUI (base) rachel@RachelsacStudio ComfyUI % git pull fatal: not a git repository (or any parent up to mount point /Volumes) Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).

Your ComfyUI dir is incomplete.
It appears that hidden directories may have been omitted during your system migration.
Reinstall them.

**Comment by KLL535** [comment]
> > > > FYI, I just installed ComfyUI Manager. First run: "ComfyUI is outdated!!" Clicking update: "Failed to update ComfyUI". Update All results in a fail too, plus a "see server log". Serverlog: "The installed ComfyUI does not have a Git repository."
> > > > (To add to this, I just followed the instructions. CMD box, cd .../custom_nodes, 'git clone https://github.com/ltdrdata/ComfyUI-Manager.git' )
> > > 
> > > 
> > > Goto `ComfyUI` dir and execute `git pull` What message is displayed?
> > > Is `ComfyUI\.git ` dir exists?
> > > `"The installed ComfyUI does not have a Git repository."` means that it is not the **ComfyUI-Manager** that is installed incorrectly, but the **ComfyUI** itself.
> > 
> > 
> > Last login: Mon Aug 19 21:34:55 on ttys000 (base) rachel@RachelsacStudio ~ % cd /Volumes/RKNavy001/ComfyUI (base) rachel@RachelsacStudio ComfyUI % git pull fatal: not a git repository (or any parent up to mount point /Volumes) Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
> 
> Your ComfyUI dir is incomplete. It appears that hidden directories may have been omitted during your system migration. Reinstall them.

Same error:
`[ComfyUI-Manager] There is no tracking branch (main)`
`ComfyUI update fail: The installed ComfyUI does not have a Git repository.`

Is there an easier way to fix it without reinstalling everything? 

The update was successful using bat files
But it doesn't work like that:

![Image](https://github.com/user-attachments/assets/e877a1d7-3341-4705-9b49-08ccc3542cda)

**Comment by ltdrdata** [comment]
> > > > > FYI, I just installed ComfyUI Manager. First run: "ComfyUI is outdated!!" Clicking update: "Failed to update ComfyUI". Update All results in a fail too, plus a "see server log". Serverlog: "The installed ComfyUI does not have a Git repository."
> > > > > (To add to this, I just followed the instructions. CMD box, cd .../custom_nodes, 'git clone https://github.com/ltdrdata/ComfyUI-Manager.git' )
> > > > 
> > > > 
> > > > Goto `ComfyUI` dir and execute `git pull` What message is displayed?
> > > > Is `ComfyUI\.git ` dir exists?
> > > > `"The installed ComfyUI does not have a Git repository."` means that it is not the **ComfyUI-Manager** that is installed incorrectly, but the **ComfyUI** itself.
> > > 
> > > 
> > > Last login: Mon Aug 19 21:34:55 on ttys000 (base) rachel@RachelsacStudio ~ % cd /Volumes/RKNavy001/ComfyUI (base) rachel@RachelsacStudio ComfyUI % git pull fatal: not a git repository (or any parent up to mount point /Volumes) Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
> > 
> > Your ComfyUI dir is incomplete. It appears that hidden directories may have been omitted during your system migration. Reinstall them.
> 
> Same error:
> `[ComfyUI-Manager] There is no tracking branch (main)`
> `ComfyUI update fail: The installed ComfyUI does not have a Git repository.`
> 
> Is there an easier way to fix it without reinstalling everything? 
> 
> The update was successful using bat files
> But it doesn't work like that:
> 
> ![Image](https://github.com/user-attachments/assets/e877a1d7-3341-4705-9b49-08ccc3542cda)

Make sure your ComfyUI-Manager is up to date. (V3.17.6)
Execute following command if the issue persists.

Goto ComfyUI dir in cmd.
```
git reset --hard
git checkout master
```
Make sure there are no error messages.

**Comment by zuoyuann** [comment]
> > FYI, I just installed ComfyUI Manager. First run: "ComfyUI is outdated!!" Clicking update: "Failed to update ComfyUI". Update All results in a fail too, plus a "see server log". Serverlog: "The installed ComfyUI does not have a Git repository."
> > (To add to this, I just followed the instructions. CMD box, cd .../custom_nodes, 'git clone https://github.com/ltdrdata/ComfyUI-Manager.git' )
> 
> Goto `ComfyUI` dir and execute `git pull` What message is displayed?
> 
> Is `ComfyUI\.git ` dir exists?
> 
> `"The installed ComfyUI does not have a Git repository."` means that it is not the **ComfyUI-Manager** that is installed incorrectly, but the **ComfyUI** itself.

I encountered the same problem today. 'ComfyUI\.git' does exist in my dir and i have tried many suggestions but no one works.

**Comment by ltdrdata** [comment]
> 
> 
> 
> 
> > > FYI, I just installed ComfyUI Manager. First run: "ComfyUI is outdated!!" Clicking update: "Failed to update ComfyUI". Update All results in a fail too, plus a "see server log". Serverlog: "The installed ComfyUI does not have a Git repository."
> > > (To add to this, I just followed the instructions. CMD box, cd .../custom_nodes, 'git clone https://github.com/ltdrdata/ComfyUI-Manager.git' )
> > 
> > Goto `ComfyUI` dir and execute `git pull` What message is displayed?
> > 
> > Is `ComfyUI\.git ` dir exists?
> > 
> > `"The installed ComfyUI does not have a Git repository."` means that it is not the **ComfyUI-Manager** that is installed incorrectly, but the **ComfyUI** itself.
> 
> I encountered the same problem today. 'ComfyUI\.git' does exist in my dir and i have tried many suggestions but no one works.


Do you use the Desktop version?
In that case, that should be fixed since ComfyUI-Manager 3.17.10.
AFAIK, latest desktop version includes fixed version of ComfyUI-Manager.

**Comment by zuoyuann** [comment]
> > > > FYI, I just installed ComfyUI Manager. First run: "ComfyUI is outdated!!" Clicking update: "Failed to update ComfyUI". Update All results in a fail too, plus a "see server log". Serverlog: "The installed ComfyUI does not have a Git repository."
> > > > (To add to this, I just followed the instructions. CMD box, cd .../custom_nodes, 'git clone https://github.com/ltdrdata/ComfyUI-Manager.git' )
> > > 
> > > 
> > > Goto `ComfyUI` dir and execute `git pull` What message is displayed?
> > > Is `ComfyUI\.git ` dir exists?
> > > `"The installed ComfyUI does not have a Git repository."` means that it is not the **ComfyUI-Manager** that is installed incorrectly, but the **ComfyUI** itself.
> > 
> > 
> > I encountered the same problem today. 'ComfyUI.git' does exist in my dir and i have tried many suggestions but no one works.
> 
> Do you use the Desktop version? In that case, that should be fixed since ComfyUI-Manager 3.17.10. AFAIK, latest desktop version includes fixed version of ComfyUI-Manager.

Thanks for your reply! I've solved it through
`git init`
`git remote add origin https://github.com/comfyanonymous/ComfyUI.git`
`git fetch --all`
`git reset --hard origin/master`
